traitsui/testing/tester/ui_wrapper.py

```diff
@@ -277,7 +285,10 @@ def _perform_or_inspect(self, interaction):
                 supported.extend(e.supported)
                 continue
             else:
-                with _event_processed():
+                context = (
+                    _event_processed if self._process_events else _nullcontext
+                )
+                with context():
                     return handler(self, interaction)
```

Comment:
why exactly is `_nullcontext` needed?  Could we just do:
```
if self._process_events:
    with _event_processed:
        return handler(self, interaction)
else:
    return handler(self, interaction)
```

Not saying we should do that even if it works.  The current code is cleaner.  I just didn't know if it was functionally necessary or not.  What exactly does a context manager that simply `yield`s do exactly?

Reply:
The code block you showed is equivalent indeed. A context manager that simply `yield`s is an no-op context manager. It is useful for situations where the context manager is optional (our exact use case here). This is why `nullcontext` is added to `contextlib` in Python, but since we are still supporting Python 3.6, I just duplicate the logic here as `_nullcontext`. Once we drop Python 3.6, it can be replaced with the one from the standard library.
See also https://bugs.python.org/issue10049

I'd prefer the noop context to repeating the code block . Here we have just one line, but in some cases, the code could be bigger. Also, if we had duplicated the code block, it may not be obvious to future developers that they need to be maintained to be exactly the same - it would be bad because it risks introducing bugs when the code block is changed in one branch and not another.